cachi2/core/package_managers/generic/models.py

```diff
@@ -66,18 +83,141 @@ def set_filename(self, info: ValidationInfo) -> "LockfileArtifact":
 
         return self
 
-    @property
-    def formatted_checksum(self) -> ChecksumInfo:
-        """Return the checksum as a ChecksumInfo object."""
-        algorithm, digest = self.checksum.split(":", 1)
-        return ChecksumInfo(algorithm, digest)
+
+class LockfileArtifactUrl(LockfileArtifactBase):
+    """
+    Defines format of a single artifact in the lockfile.
+
+    :param download_url: The URL to download the artifact from.
+    """
+
+    download_url: AnyUrl
+
+    def resolve_filename(self) -> str:
+        """Resolve the filename of the artifact."""
+        if not self.filename:
+            url_path = urlparse(str(self.download_url)).path
+            return Path(url_path).name
+        return self.filename
+
+    def get_sbom_component(self) -> Component:
+        """Return an SBOM component representation of the artifact."""
+        name = Path(self.filename).name
+        url = str(self.download_url)
+        component = Component(
+            name=name,
+            purl=PackageURL(
+                type="generic",
+                name=name,
+                qualifiers={
+                    "download_url": url,
+                    "checksum": self.checksum,
+                },
+            ).to_string(),
+            type="file",
+            external_references=[ExternalReference(url=url, type="distribution")],
+        )
+        return component
+
+
+class LockfileArtifactMavenAttributes(BaseModel):
+    """Attributes for a Maven artifact in the lockfile."""
+
+    repository_url: AnyUrl
+    group_id: str
+    artifact_id: str
+    version: str
+    classifier: str = ""
+    type: str = "jar"
+
+    @cached_property
+    def extension(self) -> str:
+        """Return the extension of the artifact."""
+        type_to_extension = {
+            "pom": "pom",
+            "jar": "jar",
+            "maven-plugin": "jar",
+            "ear": "ear",
+            "ejb": "jar",
+            "ejb-client": "jar",
+            "javadoc": "jar",
+            "javadoc-source": "jar",
+            "rar": "rar",
+            "test-jar": "jar",
+            "war": "war",
+        }
+        return type_to_extension.get(self.type, self.type)
+
+
+class LockfileArtifactMaven(LockfileArtifactBase):
+    """Defines format of a Maven artifact in the lockfile."""
+
+    type: Literal["maven"]
```

Comment:
Does MavenArtifact need type="maven" ?

Reply:
This is a bit of future proofing for the lockfile. Maven might not be the only type of artifact supported in the future ([e.g. oci](https://github.com/containerbuildsystem/cachi2/pull/692/files#diff-aadc5e0bf6d8c1eefd2742bad409f3f2fb7a684aa7cbe0d47ebfb752e07d9fd6R215)), so the artifacts should be distinguishable in the lockfile, so I have used `type`, same as in the [proposal](https://github.com/containerbuildsystem/cachi2/pull/692/).